Add SauceLabs Functional Page Loading

alex-nov-git opened the pull request:
Added: 4 independent tests in one class, that provide tests for
admin, order, frontend and customers pages. Each test checks
all pages of its type, and then checks if one or more pages crach.
TODO: remove creating customer from all tests and move it to the Fixture.

alex-nov-git committed:
Add SauceLabs Functional Page Loading

Added: 4 independent tests in one class, that provide tests for
admin, order, frontend and customers pages. Each test checks
all pages of its type, and then checks if one or more pages crach.
TODO: remove creating customer from all tests and move it to the Fixture.

re issues: merchantprotocol/PHP-Sausage-Selenium#1, merchantprotocol/relieffactor.com#357

d10nZ commented on tests/Magento_Page_Loading.php:
```diff
@@ -0,0 +1,473 @@
+        ],
+    ];
+
+    const ADMIN_USERNAME    = 'denis';
```
please replace it with **user** which will be used only for **test** purposes
such functionality should be moved into separate config.php file

jonathonbyrdziak commented on tests/Magento_Page_Loading.php:
```diff
@@ -0,0 +1,473 @@
+     */
+    private $customLayoutPages = [
+        [
+            'link' => '/doneright/enhanced/dashboard',
```
The admin url prefix `doneright` should be moved into a separate config file

d10nZ replied:
moreover we should move into separate config file all specific data and parameters which can be used by all of us (config.local.php) and by travis CI service (config.staging.php)

domain name => ''
admin domain name => ''
preferred browsers => []
admin_account => [login => 'test', pass=> 'test']
customer_account => [login => 'test', pass=> 'test']
etc

alex-nov-git replied:
I moved these links into separate configuration file. We can use this file as a global configuration for testing module. But at the same time these links may seem irrelevant in the global configuration file. The data provider method is simple method to provide arguments for test method, so we should understand that the data provider and separate links in config file is a different things. We can use links from separate config file or data provider, that returns array with hard coded links for this test case, or data provider, that returns array with links from configuration, what do you think?

jonathonbyrdziak commented on tests/Magento_Page_Loading.php:
```diff
@@ -0,0 +1,473 @@
+        3 => '/blog',
+        4 => '/ingredients',
+        5 => '/relieffactor-quickstart-pack.html',
+    ];
```
separate config file for the list of pages to test

d10nZ replied:
it is better to use dataProvider for such purposes
see: https://phpunit.de/manual/current/en/writing-tests-for-phpunit.html
   chapter: 'Data Providers'

jonathonbyrdziak commented on tests/Magento_Page_Loading.php:
```diff
@@ -0,0 +1,473 @@
+
+        if (!is_array($links) || empty($links)) {
+            $this->fail('Admin links now found');
+        }
```
should be "Admin links *not* found"

jonathonbyrdziak commented:
Looking good so far.

alex-nov-git commented on tests/Magento_Page_Loading.php:
```diff
@@ -0,0 +1,473 @@
+    }
+
+    /**
+     * Testing of pages loading in customer account
```
Fix comment

alex-nov-git committed:
Refactoring SauceLabs Functional Page Loading

Added: Configuration class which provides an ability to use local and stage configuration files.
TODO: remove creating customer from all tests and move it to the Fixture.

re issues: merchantprotocol/PHP-Sausage-Selenium#1, merchantprotocol/relieffactor.com#357

alex-nov-git committed:
Refactoring SauceLabs Functional Page Loading

Replace browsers configuration to config file
Add local config to gitignore

re issues: merchantprotocol/PHP-Sausage-Selenium#1, merchantprotocol/relieffactor.com#357
